src/openloco/openloco.cpp

```diff
@@ -883,6 +895,239 @@ namespace openloco
+                    regs2.ebp = ebx;
+                    call(0x0046DD06, regs2);
+                    ebx = regs2.ebp;
+                }
+            }
+        }
+
+        if (ebx == 0x80000000)
+        {
+            if (flags & 1)
+            {
+                return loc_4314EA();
+            }
+            else
+            {
+                game_command_nest_level--;
+                return ebx;
+            }
+        }
+
+        if ((flags & 1) == 0)
+        {
+            game_command_nest_level--;
+            return ebx;
+        }
+
+        uint16_t flagsBackup2 = _gameCommandFlags;
+        registers fnRegs2 = regs;
+        console::log("addr %d (0x%X) [EAX:%X EBX:%X ECX:%X EDX:%X EDI:%X EBP:%X]", fnRegs2.esi, addr, fnRegs2.eax, fnRegs2.ebx, fnRegs2.ecx, fnRegs2.edx, fnRegs2.edi, fnRegs2.ebp);
+        call(addr, fnRegs2);
+        uint32_t ebx2 = fnRegs2.ebx;
+        _gameCommandFlags = flagsBackup2;
+
+        if (ebx2 == 0x80000000)
+        {
+            return loc_4314EA();
+        }
+
+        if (is_editor_mode())
+        {
+            ebx = 0;
+        }
+
+        if (ebx2 < ebx)
+        {
+            ebx = ebx2;
+        }
+
+        game_command_nest_level--;
+        if (game_command_nest_level == 0)
+            return ebx;
+
+        if ((flagsBackup2 & 0x20) != 0)
+            return ebx;
+
+        call(0x0046DE2B);
+
+        if (ebx != 0 && _updating_company_id == _player_company[0])
+        {
+            _game_command_map_z = _game_command_map_z + 24;
+            call(0x0046DC9F);
+            _game_command_map_z = _game_command_map_z - 24;
+        }
+
+        return ebx;
+    }
+
+    static void sub_431908(company_id_t al, string_id dx)
+    {
+
+        registers regs;
+        regs.al = al;
+        regs.dx = dx;
+        call(0x431908, regs);
+    }
+
+    static uint32_t loc_4314EA()
+    {
+        game_command_nest_level--;
+        if (game_command_nest_level != 0)
+            return 0x80000000;
+
+        if (_updating_company_id != _player_company[0])
+            return 0x80000000;
+
+        if (_gameCommandFlags & 8)
+            return 0x80000000;
+
+        if (_9C68E6 != 0xFFFE)
+        {
+            windows::show_error(_9C68E8, _9C68E6);
+            return 0x80000000;
+        }
+
+        // advanced errors
+        if (_9C68D0 != (void*)-1)
+        {
+            auto tile = (tile_element*)_9C68D0;
+
+            switch (tile->type())
+            {
+                case element_type::unk_1: // 4
+                    _commonFormatArgs[0] = objectmgr::get<track_object>(tile->as_unk1()->track_object_id())->name;
+                    _commonFormatArgs[1] = companymgr::get(_errorCompanyId)->name;
+                    sub_431908(_errorCompanyId, string_ids::str_1421);
+                    return 0x80000000;
+
+                case element_type::unk_7: //0x1C
+                    _commonFormatArgs[0] = objectmgr::get<road_object>(tile->as_unk7()->road_object_id())->name;
```

Comment:
May yield a `nullptr`. I realise Sawyer doesn't check for this, but CI rightfully points this out as well.